File: api-code-examples/nodejs/authors-list.js
import { Iroh } from '@number0/iroh'

const node = await Iroh.memory({ enableDocs: true })
const nodeId = await node.net.nodeId()
console.log(`Started iroh node: ${nodeId}`)

const author = await node.authors.create()
console.log(`Created author: ${author.toString()}`)

const authors = await node.authors.list()
console.log('Authors:')
for (let author of authors) {
  console.log(`\t${author}`)
}

File: api-code-examples/python/authors-list.py
import iroh
import asyncio


async def main():
    # Set options to enable docs
    options = iroh.NodeOptions()
    options.enable_docs = True

    # Create in memory iroh node
    node = await iroh.Iroh.memory_with_options(options)
    node_id = await node.net().node_id()
    print(f"Started Iroh node: {node_id}")

    author = await node.authors().create()
    print(f"Created author: {author}")

    authors = await node.authors().list()
    print("Authors:")
    for auth in authors:
        print(f"\t{auth}")

# Output:
# Started Iroh node: wbwpkauwmitrwwhmw534w3u6sxyhvbivuepcdze5jd3zeqpgxyfa
# Created author: z2h3f3tozgz67g3ewvu7yxj5vtoddwoohzi2jgyrtgfl7tzocpda
# Authors:
#	fllmuelqe7lrzmlwmmydtperpafnyd6vkzukqh2l7xqg6cvjstia
#	x2ckktvt4ftjcmrb4xxmlajs7vcidkdsmxcsbd5vejezpcu6653a

asyncio.run(main())
